p2p/protocol/identify/id.go

```diff
@@ -491,13 +491,18 @@ func (ids *idService) handleIdentifyResponse(s network.Stream, isPush bool) erro
 	if !ok { // might already have disconnected
 		return nil
 	}
+	prevPushSupport := e.PushSupport
 	sup, err := ids.Host.Peerstore().SupportsProtocols(c.RemotePeer(), IDPush)
 	if supportsIdentifyPush := err == nil && len(sup) > 0; supportsIdentifyPush {
 		e.PushSupport = identifyPushSupported
 	} else {
 		e.PushSupport = identifyPushUnsupported
 	}
 	ids.conns[c] = e
+	if ids.metricsTracer != nil && e.PushSupport != prevPushSupport {
```

Comment:
Why is this conditioned on the previous push support? Why can't we just count all connections identified by push supported (yes / no)?

Reply:
I was keeping it a gauge counting support on current connections. This change will make this metric only work for new connections.